src/opensearch.ts

```diff
@@ -141,19 +151,21 @@ function extractBlock(res: ExtractedResult<WithTimestamp & Hash, OpenSearchBlock
 }
 
 function extractBalance(res: ExtractedResult<Ordinal, BalanceValue>[]): TaskEither<ApplicationError, Balance> {
-    if (res.length == 1) {
-        if (res[0].inner.length == 1) {
-            return right(Object.assign(res[0].outer, res[0].inner[0]))
-        }
-    }
-
-    return left(new ApplicationError(
-        "Not found",
-        ["Malformed data."],
-        StatusCodes.NOT_FOUND
-    ))
+    return pipe(
+        pipe(O.fromNullable(res[0]),
+            O.chain(res => pipe(
+                O.fromNullable(res.inner[0]),
+                O.map(balance => { return { ...res.outer, ...balance } })
```

Comment:
In TS world devs tend to write it like that: `O.map(balance => ({ ...res.outer, ...balance }))`. Just a convention